geopandas/array.py

```diff
@@ -228,7 +228,7 @@ def _binary_geo(op, left, right):
         # intersection can return empty GeometryCollections, and if the
         # result are only those, numpy will coerce it to empty 2D array
         data = np.empty(len(left), dtype=object)
-        data[:] = [getattr(s, op)(right) for s in left.data]
+        data[:] = [getattr(s, op)(right) if s and right else None for s in left.data]
```

Comment:
in operations like `intersection`, missing values would propagate (meaning if one of the input is missing, the output is missing)

Reply:
This is the right approach